### ISSUE TITLE
`indexes = true` in code generation should enable `tables` creation automatically

### ISSUE THREAD

**Issue by perlun** [jOOQ/jOOQ#12992]
### Steps to reproduce the problem
With the following `generate` config snippet:

```groovy
generate {
    deprecated = false
    fluentSetters = true
    immutablePojos = true
    records = false
    tables = false // <-- this is the problem
}
```

...and `indexes` using the default setting (`true`), the code generator generates an `Indexes` class which doesn't compile:

![image](https://user-images.githubusercontent.com/630613/152531605-b1e5c312-36e1-4790-b50b-a52aab62480a.png)

The reason for this is simple; from what I can tell, the `Indexes` code is dependent on `tables` being generated. In other words, just like `records` or `daos` causing `tables` to be auto-enabled, the same thing should happen for `indexes`.

Since this seems to be trivial to fix, I'll give it a look and see if I can get a fix submitted. (it's Friday, after all. :sunglasses:)

### Versions

- jOOQ: 3.16.2
- Java: Java 8
- Database (include vendor): PostgreSQL
- OS: Debian GNU/Linux
- JDBC Driver (include name if inofficial driver): `org.postgresql:postgresql`

**Comment by lukaseder** [comment]
Thanks for your report and PR. Fixed in jOOQ 3.17.0, 3.16.4 (https://github.com/jOOQ/jOOQ/issues/12995), and 3.15.8 (https://github.com/jOOQ/jOOQ/issues/12996)

**Comment by lukaseder** [comment]
The suggested fix has introduced a curious regression in a different code generation test:

```xml
<generate>
    <relations>false</relations>
    <tables>false</tables>
    <routines>false</routines>
    <udts>false</udts>
    <sequences>false</sequences>
    <links>false</links>
    <queues>false</queues>
    <records>false</records>
    <pojos>true</pojos>
</generate>
```

This configuration is supposed to generate POJOs only, but now also generates tables because it doesn't specifically configure `<indexes>` (defaulting to `true`). The test generates a table without any index, so enabling the `<tables>` flag in this edge case seems wrong

**Comment by lukaseder** [comment]
Edge case fixed and backported again to jOOQ 3.17.0, 3.16.4 (https://github.com/jOOQ/jOOQ/issues/12995), and 3.15.8 (https://github.com/jOOQ/jOOQ/issues/12996)
